pkg/server/server_test.go

```diff
@@ -462,3 +472,71 @@ func TestListObjects_UnhappyPaths(t *testing.T) {
 		require.ErrorIs(t, err, serverErrors.NewInternalError("", errors.New("error reading from storage")))
 	})
 }
+
+func TestObosolete(t *testing.T) {
+	ctx := context.Background()
+	logger := logger.NewNoopLogger()
+	transport := gateway.NewNoopTransport()
+	store := ulid.Make().String()
+	modelID := ulid.Make().String()
+
+	mockController := gomock.NewController(t)
+	defer mockController.Finish()
+
+	data, err := os.ReadFile("testdata/github.json")
+	require.NoError(t, err)
+
+	var gitHubTypeDefinitions openfgapb.WriteAuthorizationModelRequest
+	err = protojson.Unmarshal(data, &gitHubTypeDefinitions)
+	require.NoError(t, err)
+
+	mockDatastore := mockstorage.NewMockOpenFGADatastore(mockController)
+
+	mockDatastore.EXPECT().ReadAuthorizationModel(gomock.Any(), store, modelID).AnyTimes().Return(&openfgapb.AuthorizationModel{
+		SchemaVersion:   typesystem.SchemaVersion1_0,
+		TypeDefinitions: gitHubTypeDefinitions.GetTypeDefinitions(),
+	}, nil)
+	mockDatastore.EXPECT().ListObjectsByType(gomock.Any(), store, "repo").AnyTimes().Return(nil, errors.New("error reading from storage"))
+
+	s := Server{
+		datastore: mockDatastore,
+		transport: transport,
+		logger:    logger,
+		config: &Config{
+			ResolveNodeLimit:           25,
+			ListObjectsDeadline:        5 * time.Second,
+			ListObjectsMaxResults:      1000,
+			AllowEvaluating1Dot0Models: false,
+			AllowWriting1Dot0Models:    false,
+		},
+	}
+
+	t.Run("throw_obsolete_error_in_check", func(t *testing.T) {
+		_, err = s.Check(ctx, &openfgapb.CheckRequest{
+			StoreId:              store,
+			AuthorizationModelId: modelID,
+			TupleKey: tuple.NewTupleKey(
+				"team:abc",
+				"member",
+				"user:anne"),
+		})
+		require.Error(t, err)
+		e, ok := status.FromError(err)
+		require.True(t, ok)
+		require.Equal(t, codes.Code(openfgapb.ErrorCode_invalid_authorization_model), e.Code())
+	})
+
+	t.Run("throw_obsolete_error_in_listobject", func(t *testing.T) {
```

Comment:
Want to add a nested test for the other endpoints as well? That way we have server test coverage for `Check`, `ListObjects`, `Expand`, `Write`, and `WriteAssertions`.

I know we have `commands` tests in Expand and Write etc.. But testing at server layer is higher fidelity, because we may have different return behavior that happens upstream from the command.

Reply:
done